### PR TITLE
fix bug with delete expression

### PR DESCRIPTION
**Pull request by stp59**
title

### CODE REVIEW

File: resources/regressions/all.json/gambit_results.json [hunk 1]
```diff
@@ -309,156 +309,163 @@
   },
   {
     "description": "DeleteExpressionMutation",
-    "diff": "--- original\n+++ mutant\n@@ -6,7 +6,8 @@\n \n     function myIdentity(uint256 x) public pure returns (uint256) {\n \tuint256 result = 0;\n-\tfor (uint256 i = 0; i < x; i++) {\n+\t/// DeleteExpressionMutation(`i++` |==> `/* i++ */`) of: `for (uint256 i = 0; i < x; i++) {`\n+\tfor (uint256 i = 0; i < x; /* i++ */) {\n \t    result ++;\n \t}\n \treturn result;\n",
```
Comment (chandrakananandi) [inline resources/regressions/all.json/gambit_results.json:312]: what are all the changes in this file? Shouldn't it only add new mutants?

File: resources/regressions/all.json/mutants/47/IfStatementMutation/IfStatementMutation.sol [hunk 2]
```diff
@@ -4,8 +4,8 @@ pragma experimental ABIEncoderV2;
 
 contract IfStatementMutation {
     function myBooleanNegation(bool a) public pure returns (bool) {
-	/// IfStatementMutation(`a` |==> `false`) of: `if (a) {`
```
Comment (chandrakananandi) [inline resources/regressions/all.json/mutants/47/IfStatementMutation/IfStatementMutation.sol:7]: why is this changing?

File: resources/regressions/all.json/mutants/48/IfStatementMutation/IfStatementMutation.sol [hunk 3]
```diff
@@ -4,8 +4,8 @@ pragma experimental ABIEncoderV2;
 
 contract IfStatementMutation {
     function myBooleanNegation(bool a) public pure returns (bool) {
-	/// IfStatementMutation(`a` |==> `true`) of: `if (a) {`
```
Comment (chandrakananandi) [inline resources/regressions/all.json/mutants/48/IfStatementMutation/IfStatementMutation.sol:7]: same, why did this change?

File: resources/regressions/all.json/mutants/49/SwapArgumentsOperatorMutation/SwapArgumentsOperatorMutation.sol [hunk 4]
```diff
@@ -4,12 +4,12 @@ pragma experimental ABIEncoderV2;
 
 contract SwapArgumentsOperatorMutation {
     function mySubtraction(uint256 x, uint256 y) public pure returns (uint256) {
```
Comment (chandrakananandi) [inline resources/regressions/all.json/mutants/49/SwapArgumentsOperatorMutation/SwapArgumentsOperatorMutation.sol:6]: why changed?

**Review by chandrakananandi — CHANGES_REQUESTED** [review]
Looks good. I just don't understand why so many other tests had to be changed.

File: README.md [hunk 5]
```diff
@@ -416,7 +416,7 @@ Gambit implements the following mutation operators
 | **unary-operator-mutation**          | Replace a unary operator with another                    | `~a` -> `-a`                                   |
 | **require-mutation**                 | Alter the condition of a `require` statement             | `require(some_condition())` -> `require(true)` |
 | **assignment-mutation**              | Replaces the rhs of a mutation                           | `x = foo();` -> `x = -1;`                      |
-| **delete-expression-mutation**       | Comment out an expression statement                      | `foo();` -> `/* foo() */;`                     |
+| **delete-expression-mutation**       | Replace an expression statement with `assert(true)`      | `foo();` -> `assert(true);`                    |
```
Comment (chandrakananandi) [inline README.md:419]: Also let's fix all the typos in the readme.

**Review by chandrakananandi — CHANGES_REQUESTED** [review]
ok let's also add include-paths. I have wanted that for a while anyway.

File: resources/regressions/all.json/gambit_results.json (hunk 1 above)
Comment (stp59) [reply resources/regressions/all.json/gambit_results.json:312]: @chandrakananandi these were changed automatically by a script i ran to update the regressions. im not sure why this happened or if it is necessary. should we check in with Ben?

Comment (chandrakananandi) [reply resources/regressions/all.json/gambit_results.json:312]: @BenTheKush is this expected?

File: README.md [hunk 6]
```diff
@@ -149,32 +149,33 @@ are passed directly through to the solc compiler.
...
+  gambit mutate --filename path/to/file.sol --solc_base_path base/path/dir/.
   ```
 
 * **Specify remappings:** To indicate where Solidity should find libraries,
   use solc's [import remapping][remapping] syntax with `--solc_remappings`:
 
   ```bash
-  gambit mutate path/to/file.sol \
+  gambit mutate --filename path/to/file.sol \
     --solc_remapping @openzepplin=node_modules/@openzeppelin @foo=node_modules/@foo
```
Comment (chandrakananandi) [inline README.md:163]: fix this typo

Comment (chandrakananandi) [reply README.md:163]: it's `solc_remappings`

File: README.md [hunk 7]
```diff
@@ -401,11 +402,12 @@ This has the following structure:
...
-| :------------------- | :---------------------------------------------------------------------------- |
-| `--solc_base_path`   | passes a value to solc's `--base-path` argument                               |
-| `--solc_allow_paths` | passes a value to solc's `--allow-paths` argument                             |
-| `--solc_remapping`   | passes a value to directly to solc: this should be of the form `prefix=path`. |
+| Option                | Description                                                                   |
+| :-------------------- | :---------------------------------------------------------------------------- |
+| `--solc_base_path`    | passes a value to solc's `--base-path` argument                               |
+| `--solc_allow_paths`  | passes a value to solc's `--allow-paths` argument                             |
+| `--solc_include_path` | passes a value to solc's `--include-path` argument                           |
+| `--solc_remapping`    | passes a value to directly to solc: this should be of the form `prefix=path`. |
```
Comment (chandrakananandi) [inline README.md:410]: typo

File: README.md [hunk 8]
```diff
@@ -401,11 +402,12 @@ This has the following structure:
...
-| Option               | Description                                                                   |
-| :------------------- | :---------------------------------------------------------------------------- |
-| `--solc_base_path`   | passes a value to solc's `--base-path` argument                               |
-| `--solc_allow_paths` | passes a value to solc's `--allow-paths` argument                             |
-| `--solc_remapping`   | passes a value to directly to solc: this should be of the form `prefix=path`. |
+| Option                | Description                                                                   |
+| :-------------------- | :---------------------------------------------------------------------------- |
+| `--solc_base_path`    | passes a value to solc's `--base-path` argument                               |
+| `--solc_allow_paths`  | passes a value to solc's `--allow-paths` argument                             |
+| `--solc_include_path` | passes a value to solc's `--include-path` argument                           |
```
Comment (chandrakananandi) [inline README.md:409]: can you add an example to show this usage?

**Review by chandrakananandi — CHANGES_REQUESTED** [review]
Mostly there!

File: resources/regressions/all.json/gambit_results.json (hunk 1 above)
Comment (BenTheKush) [reply resources/regressions/all.json/gambit_results.json:312]: You can see that this makes sense: the first weird diff is for a delete expression mutation. 
* It looks like we are addingthe first new `DeleteExpressionmutation` at mutant id = 45 (which was already a delete expression mutation).
* Then the _old_ `DeleteExpressionMutant` becomes mutant id 46, which used to be `IfStatementMutation`.
* The displaced `IfStatementMutation`, which was mid=46, becomes mid=47, which was also an `IfStatementMutation`.
* The second displaced `IfStatementMutation` becomes mid=48, which was a `SwapArgumentMutation`. This now falls to mid=49, etc, etc

File: resources/regressions/all.json/mutants.log [hunk 9]
```diff
@@ -42,25 +42,26 @@
...
+58,SwapArgumentsOperatorMutation,SwapArgumentsOperatorMutation/SwapArgumentsOperatorMutation.sol,43:9,x >> y,y >> x
+59,UnaryOperatorMutation,UnaryOperatorMutation/UnaryOperatorMutation.sol,7:9,~,++
+60,UnaryOperatorMutation,UnaryOperatorMutation/UnaryOperatorMutation.sol,7:9,~,--
+61,UnaryOperatorMutation,UnaryOperatorMutation/UnaryOperatorMutation.sol,11:9,++,--
+62,UnaryOperatorMutation,UnaryOperatorMutation/UnaryOperatorMutation.sol,11:9,++,~
+63,UnaryOperatorMutation,UnaryOperatorMutation/UnaryOperatorMutation.sol,15:9,--,++
+64,UnaryOperatorMutation,UnaryOperatorMutation/UnaryOperatorMutation.sol,15:9,--,~
+65,UnaryOperatorMutation,UnaryOperatorMutation/UnaryOperatorMutation.sol,19:3,++,--
+66,UnaryOperatorMutation,UnaryOperatorMutation/UnaryOperatorMutation.sol,24:3,--,++
+67,ElimDelegateMutation,ElimDelegateMutation/ElimDelegateMutation.sol,25:55,delegatecall,call
```
Comment (BenTheKush) [inline resources/regressions/all.json/mutants.log:67]: @chandrakananandi @stp59 this is a much cleaner view of what the messy diffs are trying to display.

File: resources/regressions/all.json/gambit_results.json (hunk 1 above)
Comment (stp59) [reply resources/regressions/all.json/gambit_results.json:312]: we just spoke over slack. these are expected/necessary and we should keep them

File: resources/regressions/all.json/mutants.log (hunk 9 above)
Comment (stp59) [reply resources/regressions/all.json/mutants.log:67]: got it thanks

File: resources/regressions/all.json/gambit_results.json (hunk 1 above)
Comment (chandrakananandi) [reply resources/regressions/all.json/gambit_results.json:312]: sg

**Review by BenTheKush — APPROVED** [review]
LGTM!

**Review by chandrakananandi — APPROVED** [review]
:shipit: thank you!

**Pull request merged**